app/src/services/Activity.js

```diff
@@ -0,0 +1,19 @@
+import api from '../services/Api';
+
+class ActivityService {
+    async getActivityList(id, coords, categoryId, activityId) {
+        const { longitude, latitude } = coords;
+        const url =
+            `/activity/list?id=${id}&coords=${longitude},${latitude}` +
+            (categoryId ? `&categoryId=${categoryId}` : '') +
+            (activityId ? `&activityId=${activityId}` : '');
```

Comment:
Só uma ideia para evitar ficar usando ternário para parâmetro, existe um helper do javascript voltado para fazer query param.
```suggestion
  const queryParams = new URLSearchParams({
      id,
      coords: `${longitude},${latitude}`,
      ...(categoryId && { categoryId }),
      ...(activityId && { activityId }),
    });
        const url =
            `/activity/list?${queryParams.toString()}`
```

Reply:
feito